Vikta-API-testing-sample/src/test/java/com/test/api/category/CategoryTest.java

```diff
@@ -2,53 +2,65 @@
 
 import com.test.api.dto.CategoryDTO;
 import com.test.api.execution.BaseTest;
-import org.junit.jupiter.api.Test;
-
-import java.util.Arrays;
+import io.restassured.response.Response;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 import static io.restassured.RestAssured.given;
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.*;
+import static org.assertj.core.api.Assertions.*;
 
-/**
- * Test for Category API
- */
 public class CategoryTest extends BaseTest {
 
-    /**
-     * Test ob category title verification
-     */
+    private final List<Integer> idCategoryToClean = new ArrayList<>();
+    private final String idQueryParam = "id";
+    private static final Logger logger = LogManager.getLogger(CategoryTest.class);
+
+    @AfterEach
+    void cleanup(TestInfo testInfo) {
+        if (testInfo.getTags().contains("needs-cleanup")) {
+            for(int id : idCategoryToClean) {
+                given().spec(defaultRequestSpec())
+                        .queryParam(idQueryParam, id)
+                        .when()
+                        .delete(getTestEnvironment().getCategoryPath());
+            }
+        }
+    }
+
+
     @Test
-    public void verifyGetCategory() {
+    public void canGetCategoryByCorrectId() {
         final int categoryId = 5;
         final String expectedTitle = "Street";
 
-        assertThat("Category title is unexpected",
-                getExistingCategoryById(categoryId).getTitle(),
-                equalTo(expectedTitle));
+        assertThat(getExistingCategoryById(categoryId).getTitle())
+                .as(() -> String.format("Category title should be %s, but is %s",getExistingCategoryById(categoryId).getTitle(), expectedTitle))
```

Comment:
You don't have to pass the message as lambda expression, it can be String. Same for all messages below